shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/presentation/viewmodel/handler/feeding/DefaultFeedingHandler.kt

```diff
@@ -18,62 +19,90 @@ import com.epmedu.animeal.feeding.presentation.event.FeedingEvent.Finish
 import com.epmedu.animeal.router.presentation.RouteHandler
 import com.epmedu.animeal.timer.presentation.handler.TimerHandler
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-class DefaultFeedingHandler @Inject constructor(
-    stateDelegate: StateDelegate<FeedingPointState>,
+@Suppress("LongParameterList", "TooManyFunctions")
+class DefaultFeedingHandler(
+    stateDelegate: StateDelegate<FeedState>,
     actionDelegate: ActionDelegate,
     routeHandler: RouteHandler,
     errorHandler: ErrorHandler,
     feedingPointHandler: FeedingPointHandler,
     timerHandler: TimerHandler,
     private val fetchCurrentFeedingPointUseCase: FetchCurrentFeedingPointUseCase,
+    private val fetchFeedingPointByIdUseCase: FetchFeedingPointByIdUseCase,
+    private var getFeedStateUseCase: GetFeedStateUseCase,
     private val startFeedingUseCase: StartFeedingUseCase,
     private val cancelFeedingUseCase: CancelFeedingUseCase,
     private val rejectFeedingUseCase: RejectFeedingUseCase,
     private val finishFeedingUseCase: FinishFeedingUseCase,
-    private val forcedArgumentsUseCase: ForcedArgumentsUseCase
 ) : FeedingHandler,
-    StateDelegate<FeedingPointState> by stateDelegate,
+    StateDelegate<FeedState> by stateDelegate,
     ActionDelegate by actionDelegate,
     FeedingPointHandler by feedingPointHandler,
     RouteHandler by routeHandler,
     TimerHandler by timerHandler,
     ErrorHandler by errorHandler {
 
-    override suspend fun fetchCurrentFeeding() {
-        val forcedFeedingPointId = forcedArgumentsUseCase<String>(FORCED_FEEDING_POINT_ID, hashCode())
-        if (forcedFeedingPointId != null) return
-
-        fetchCurrentFeedingPointUseCase()?.let { feedingPoint ->
-            showSingleReservedFeedingPoint(FeedingPointModel(feedingPoint))
-            startRoute()
+    override var feedingStateFlow: StateFlow<FeedState> = stateFlow
+    override fun CoroutineScope.fetchCurrentFeeding() {
+        launch {
+            fetchCurrentFeedingPointUseCase()?.let { feedingPoint ->
+                updateState {
+                    copy(
+                        feedPoint = FeedingPointModel(feedingPoint),
+                        feedingConfirmationState = FeedingConfirmationState.FeedingStarted
+                    )
+                }
+                showSingleReservedFeedingPoint(FeedingPointModel(feedingPoint))
+                startRoute()
+            } ?: run {
+                updateState {
+                    copy(
+                        feedingConfirmationState = FeedingConfirmationState.Dismissed
+                    )
+                }
+            }
         }
+        stateFlow.onEach {
+            // updates state flow at repository
+            getFeedStateUseCase().emit(it.toDomainFeedState())
```

Comment:
Why did you decide to create a collector instead of just put updating feed in the repository from updateFeedingState method?

Reply:
`updateFeedingState` avoids code duplication but doesn't prevent `feedingConfirmationState` of being updated out of handler (at any view model for example).